api/src/main/java/org/openmrs/module/ModuleUtil.java

```diff
@@ -571,7 +577,16 @@ public static void expandJar(File fileToExpand, File tmpModuleDir, String name,
 					String entryName = jarEntry.getName();
 					// trim out the name path from the name of the new file
 					if (!keepFullPath && name != null) {
-						entryName = entryName.replaceFirst(name, "");
+						// if 'name' is a file then make sure that only path is trimmed out and not the file name itself 
+						if (name.equals(entryName)) {
+							int lastSlash = name.lastIndexOf('/');
+							if (lastSlash >= 0) {
+								entryName = entryName.replaceFirst(name.substring(0, lastSlash + 1), "");
```

Comment:
The failing test was expanding file without full path: `expandJar_shouldExpandFileWithoutParentTreeIfNameIsFileAndKeepFullPathIsFalse()`. In this case `name` provides filename with path e.g. FOLDER1/FOLDER2/FILE. Original function was stripping this whole string from jar filename which was leaving it empty. As a result nothing was getting extracted. What I did is to make sure in that specific case that only path part of `name` gets trimmed out, so that jar filename still stays and gets extracted.